### ISSUE TITLE
Events that work only on the client side.

### ISSUE THREAD

**Issue by blinkuz** [AscensionGameDev/Intersect-Engine#863]
**Is your feature request related to a problem? Please describe.**
I do not know whether to mention this as a problem but currently it is known by all that the current event system is useful only for certain processes where they do not need to execute extensive logic due to this some of the devs were talking about a possible functionality that can be used to take advantage of what has been done with some types of events.

**Describe the solution you'd like**
The idea that has emerged is to have events that work in a client context side, there are certain processes that do not need verification by the server and that can easily be delegated only to the client.

And taking into account all of the above and respecting the community's creed that suggestions or PR for general things and not specific to a type of game, I want to mention some basic use cases:

- Change of sprites for elements of the game only visible to the player, imagine showing a reflection of a silhouette of the player when he approaches the edge of a river / ocean (this specific example I do not know if it is the most correct but it is from where the idea came up).
- Play animations or sound effects for the player this can be linked to many processes or notifications of certain activities only to the user.
- Npcs only on the client side, it is true that there are plans to create the instance system which may be a better solution for the next use case but I was thinking a lot about how it would be the best way to simulate a type of cinematic to my own project, although it is something very simple of a moving npc that shows a basic dialog to the player but that other users do not see.

**Additional context**
Obviously this is just an idea that I would like to comment on here and see what the others involved in the project think, another point to take into account is that the number of events that would be added to this "modality" would be limited but taking advantage of the conditionals and other control elements that make events a good tool these days.

**Comment by jcsnider** [comment]
This would have to be beyond basic. Even stuff like variable conditions would not work because the client never knows its own or global variables values. 

The silhouette thing should be custom coded, or if it were it could be more efficient graphically and you could do actual reflections.

Instancing will help with solo cutscenes and such. 

Regardless I totally see the appeal of this, but it's not something I will purpose. Work required does not justify the benefits in my opinion.

**Comment by Arufonsu** [comment]
> 
>     * Npcs only on the client side, it is true that there are plans to create the instance system which may be a better solution for the next use case but I was thinking a lot about how it would be the best way to simulate a type of cinematic to my own project, although it is something very simple of a moving npc that shows a basic dialog to the player but that other users do not see.
>
I wouldn't personally move values such as NPCs nor Events to the client side, they could easily be exploited that way on some games, trusting the client that much isn't particularly safe. Having instances within the game doesn't means that everything happening there will not be checked by the server, that wouldn't be safe at all either. It may be helpful for things such as "cutscenes", YES. but then this is currently not a feature with all its words, and its probably too niche, or probably not? not up to me 😆 !


> 
>     * Play animations or sound effects for the player this can be linked to many processes or notifications of certain activities only to the user.
>
I would (and i actually did) move few values that don't really need to be checked in the server side all the time, like the number of walking frames, among with other minor things that are only intended to be a graphical thing only to be seen in the local client side and could in no way lead to any form of exploit in the server side. In this one example, if an individual came over to change the speed or number of walking frames, they would simply see them behaving differently on their side, and would have no real outcome in the server, thus, what others see.

> 
>     * Change of sprites for elements of the game only visible to the player, imagine showing a reflection of a silhouette of the player when he approaches the edge of a river / ocean (this specific example I do not know if it is the most correct but it is from where the idea came up).
> 
Regardless this one, there's few guides around on how to do this with any monogame related game project, it can't be an impossible thing, i saw an intersect project on forums where they generate a sort of "shadow" behind their sprites.

**Comment by blinkuz** [comment]
> I wouldn't personally move values such as NPCs nor Events to the client side, they could easily be exploited that way on some games

Obviously, it is not delegating these events to the client, it is creating new types of events with their own logic but reusing much of the work already done, in the case of npcs it would be a new type of npc completely separate from the existing ones and with very functions. limited as I said, an npc with a dialog would be something very basic but hahaha this is just an idea that will probably never be implemented officially.
